### PR TITLE
Make publish do less work

### PR DESCRIPTION
**Pull request by davidfowl**
- Added a runtime switch to build and compile
- Flow the runtime from publish through build and compile

### CODE REVIEW

File: src/dotnet-build/CompileContext.cs [hunk 1]
```diff
@@ -275,15 +275,17 @@ private bool InvokeCompileOnDependency(ProjectDescription projectDependency)
 
             args.Add("--framework");
             args.Add($"{projectDependency.Framework}");
+            
+            if (!string.IsNullOrEmpty(_args.RuntimeValue))
```
Comment (cdmihai) [inline src/dotnet-build/CompileContext.cs]: Refactoring idea: Add `ConstructArgumentString` methods to CompilerCommandApp to encapsulate how this string is formed. Then delete the 2 methods from CompileContext

File: src/dotnet-compile/Program.cs [hunk 2]
```diff
@@ -330,35 +328,28 @@ private static bool CompileProject(ProjectContext context, CompilerCommandApp ar
...
-            if (success && !args.NoHostValue && compilationOptions.EmitEntryPoint.GetValueOrDefault())
-            {
-                generateBindingRedirects = true;
-                var rids = PlatformServices.Default.Runtime.GetAllCandidateRuntimeIdentifiers();
-                var runtimeContext = ProjectContext.Create(context.ProjectDirectory, context.TargetFramework, rids);
-                runtimeContext
-                    .MakeCompilationOutputRunnable(outputPath, args.ConfigValue);
-            }
-            else if (!string.IsNullOrEmpty(context.ProjectFile.TestRunner))
+            if (success)
```
Comment (cdmihai) [inline src/dotnet-compile/Program.cs]: Can you extract all the code beyond this point into a separate method called something like `PublishForLocalMachine`? Ideally, for the long term changes I'd want Compile to just call the compilers and nothing else :)

File: src/dotnet-build/CompileContext.cs (hunk 1 above)
Comment (davidfowl) [inline src/dotnet-build/CompileContext.cs]: Not doing that as part of this change. File an issue for refactoring.

File: src/Microsoft.DotNet.Compiler.Common/ProjectContextExtensions.cs [hunk 3]
```diff
@@ -34,21 +35,28 @@ public static void MakeCompilationOutputRunnable(this ProjectContext context, st
 
             if (context.TargetFramework.IsDesktop())
             {
+                // On full framework, copy all dependencies to the output path
```
Comment (livarcocc) [inline src/Microsoft.DotNet.Compiler.Common/ProjectContextExtensions.cs]: I think it would be great to encapsulate these comments into methods.

**Comment by livarcocc** [comment]
Small comment, LGTM.

**Comment by livarcocc** [comment]
I have merged these changes with #1138. Closing this PR.

**Pull request closed without merging**